src/Waives.Pipelines/Pipeline.cs

```diff
@@ -49,20 +50,23 @@ namespace Waives.Pipelines
     /// }
     /// ]]>
     /// </example>
-    public class Pipeline : IObservable<WaivesDocument>
+    public class Pipeline
     {
         private readonly IHttpDocumentFactory _documentFactory;
-        private IObservable<WaivesDocument> _pipeline = Observable.Empty<WaivesDocument>();
-        private Action _onPipelineCompleted;
+        private readonly int _maxConcurrency;
+        private IObservable<Document> _docSource = Observable.Empty<Document>();
+        private Action _onPipelineCompletedUserAction = () =>{};
         private readonly Action<DocumentError> _onDocumentError;
         private Action<DocumentError> _userErrorAction = err => { };
-        private readonly IRateLimiter _rateLimiter;
         private readonly ILogger _logger;
 
-        internal Pipeline(IHttpDocumentFactory documentFactory, IRateLimiter rateLimiter, ILogger logger)
+        private readonly List<Func<WaivesDocument, Task<WaivesDocument>>> _docActions =
```

Comment:
💪 The `Func<WaivesDocument, Task<WaivesDocument>>` type is used in a few places around the library. I think we should create a delegate with this signature so that it can have a meaningful name.